docker/Dockerfile

```diff
@@ -69,6 +69,25 @@ RUN set -e; \
         esac)"; \
     . <(sed 's/^\(..*\)/export DISTRO_\1/' '/etc/os-release'); \
     case "$DISTRO_ID-$DISTRO_VERSION_ID" in \
+    'debian-12') \
+        export DEB_SRC_LIST=/etc/apt/sources.list; \
+        [ -e "$DEB_SRC_LIST" ] || \
+        echo 'deb http://deb.debian.org/debian bookworm main contrib non-free'                        >> "$DEB_SRC_LIST" && \
```

Comment:
Don't use &&, there's set -e

Reply:
We need `&&` because all of them come after the first `-e` check.

Reply:
We print all of these to the source list only if the source list itself nonexists. You can treat them as a workaround and if Debian 12 add them back we can switch to add-apt-respository

Reply:
> We print all of these to the source list only if the source list itself nonexists. You can treat them as a workaround and if Debian 12 add them back we can switch to add-apt-respository

You're probably replying to another thread.

They are unlikely to move it back if there's already an intentional replacement.
What I'm saying is, you'll have to delete debian's own lines if you added yours, but to me that's not a good idea.

Reply:
They point to the same URLs. So I am not too worry about it.

Reply:
- Only the URL parts are shared, the rest are not, one is still shadowing the others.
- Tooling would be confusing, e.g. if the .sources shadows .list, add-apt-repo may complete successfully but has no effect, and vice versa.

Reply:
I am not assuming && > ||

It is just `if xxx then yy && zz && aa`. It is purely sequential.

Reply:
You're writing the 2nd line regardless of the file existence

Reply:
> * Tooling would be confusing, e.g. if the .sources shadows .list, add-apt-repo may complete successfully but has no effect, and vice versa.

That is not because of our change. `add-apt-repository` has already succeeded without effect. The is the tooling problem.

Reply:
that's so called early adopter tax where we discuss about this. Upstream should figure it out for us.

Reply:
Ack. But I think it is OK for now since I only want to try to build our code on it. If there are any time-consuming issues we can always stop using it.